deep_quoridor/src/utils/timer.py

```diff
@@ -0,0 +1,83 @@
+import time
+from typing import Optional
+
+import wandb
+import wandb.wandb_run
+
+from utils.misc import format_time
+
+
+class Timer:
+    """
+    Class to measure elapsed time and log it to the console and optionally wandb.
+    It's a class just to create a container for the functions and variables, it can't be instantiated.
+    TO DO: would be nice if it would easily support multi-process, so we can gather information coming
```

Comment:
I think that the "right" way to do multiprocess logging with wandb is to have a separate wandb run for each process, and have them all be in the same wandb "group". I prototyped that a while back and it works well, but i didn't PR it and our code has changed quite a bit since then. It's a bit ugly now that we start new worker processes for each self-play session - that will generate a lot of runs. But if we fix the memory leak then it would work well. It's pretty easy to generate graphs that show all the info for all runs in the wandb group.